alphabase/constants/atom.py

```diff
@@ -200,3 +204,195 @@ def calc_mass_from_formula(formula: str):
         mass of the formula
     """
     return np.sum([CHEM_MONO_MASS[elem] * n for elem, n in parse_formula(formula)])
+
+
+class ChemicalCompositonFormula:
+    """
+    Initialize the ChemicalCompositonFormula with a given formula.
+
+    Parameters
+    ----------
+    formula : str
+        The chemical formula as a string.
+
+    Returns
+    -------
+    None
+    """
+
+    def __init__(self, formula=None):
+        self.elements = (
+            defaultdict(int) if formula is None else self._parse_formula(formula)
+        )
+
+    @classmethod
+    def from_smiles(cls, smiles: str) -> "ChemicalCompositonFormula":
+        """
+        Create a ChemicalCompositonFormula instance from a SMILES string.
+
+        Parameters
+        ----------
+        smiles : str
+            The SMILES representation of the molecule.
+
+        Returns
+        -------
+        ChemicalCompositonFormula
+            An instance of the class based on the SMILES string.
+
+        Raises
+        ------
+        ValueError
+            If the SMILES string is invalid and can't be converted to an RDKit molecule.
+        """
+        mol = Chem.MolFromSmiles(smiles)
+        if not mol:
+            raise ValueError(f"Invalid RDKit molecule: {smiles}")
+        formula = rdMolDescriptors.CalcMolFormula(
+            mol, separateIsotopes=True, abbreviateHIsotopes=False
+        )
+        formula = formula.replace("[1H]", "H")
```

Comment:
could you add a comment why this is required? is this a common substitution? is it proteomics- or alphabase-specific?

Reply:
If 'abbreviateHIsotopes=True' we get a D instead of [2H] and the modification.tsv expects [2H]. As a side effect we now get [1H] instead of regular hydrogens, so I gotta substitute them back.